profiles/common/modules/custom/tmgmt_poetry/tmgmt_poetry.module

```diff
@@ -586,3 +660,80 @@ function tmgmt_poetry_page_alter(&$page) {
     unset($page['content']['workbench_block']);
   }
 }
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ *
+ * Change tmgmt_entity_ui_translate_form form to
+ * allow user to request a translation update.
+ */
+function tmgmt_poetry_form_tmgmt_entity_ui_translate_form_alter(&$form, &$form_state) {
+  $entity_type = $form_state['entity_type'];
+  if ($entity_type == 'node') {
+    $node = $form_state['entity'];
+    if (workbench_moderation_node_moderated($node)) {
+
+      $existing_jobs = tmgmt_job_item_load_latest(
+        'workbench_moderation',
+        $entity_type,
+        $node->nid,
+        $node->translations->original
+      );
+
+      if (!empty($existing_jobs)) {
+
+        $translation_possible = TRUE;
+
+        // Check the poetry status of subjobs.
+        foreach ($existing_jobs as $existing_job) {
+          // Check if it's a poetry job.
+          $translator = $existing_job->getTranslator();
+          if ($translator && $translator->plugin != 'poetry') {
+            return;
+          }
+
+          // Check the poetry status.
+          $result = db_select('poetry_status', 'ps')
+            ->fields('ps')
+            ->condition('tjiid', $existing_job->tjiid, '=')
+            ->execute()
+            ->fetchAssoc();
+
+          if (isset($result['status']) && $result['status'] == POETRY_STATUS_MESSAGE_PEN) {
+            $translation_possible = FALSE;
+          }
+        }
+
+        // Force form to allow user actions.
+        $form_state['force_actions'] = TRUE;
+
+        if ($translation_possible) {
+          // Add the new submit button.
+          $form['actions']['request'] = [
+            '#type' => 'submit',
+            '#value' => t('Request translation update'),
+            '#submit' => ['tmgmt_workbench_translate_form_submit'],
+            '#validate' => ['tmgmt_workbench_translate_form_validate'],
+          ];
+        }
+        else {
+          // Disable all actions.
+          unset($form['actions']);
+          $lang_codes = array_keys($form['languages']['#options']);
+          foreach ($lang_codes as $lang_code) {
+            $form['languages'][$lang_code] = [
+              '#type' => 'checkbox',
+              '#disabled' => TRUE,
+            ];
+          }
+          $form['languages_description'] = [
```

Comment:
Is it not possible here to use drupal_set_message? The result seems to be the same, but using Form API for that is a bit overcomplicated solution, isn't it?